webapp/src/main/resources/public/js/ict/Ict.js

```diff
@@ -80,25 +85,46 @@ class Ict {
         return string;
     }
 
-    wrapNode(node, onClick) {
+    wrapNode(node, onClick, addMTCSS) {
         var textUnits = IctMetadataExtractor.getTextUnits(this.getStringFromNode(node));
 
         if (node.nodeType === Node.TEXT_NODE) {
             node = node.parentNode;
         }
 
+
+        var mtRequired = this.mtEnabled && textUnits[0]['isTranslated'] === 'false';
+        var eventListenerClassName = mtRequired ? "mojito-ict-string-mt" : "mojito-ict-string-active"
+
         if (!node.classList.contains("mojito-ict-string")) {
-           try {
+            try {
                 node.className += " mojito-ict-string";
             } catch (e) {
             }
 
             node.addEventListener("mouseenter", (e) => {
-                e.target.classList.add("mojito-ict-string-active");
+                e.target.classList.add(eventListenerClassName);
             });
 
             node.addEventListener("mouseleave", (e) => {
-                e.target.classList.remove("mojito-ict-string-active");
+                e.target.classList.remove(eventListenerClassName);
+            });
+        }
+
+        if (mtRequired) {
+            chrome.runtime.sendMessage(textUnits[0], function (response) {
+                if (response.data && node.textContent !== response.data) {
+                    if (node.nodeType === Node.ELEMENT_NODE && node.getAttribute('placeholder')) {
+                        node.setAttribute('placeholder', response.data)
+                    } else {
+                        textUnits[0]['isMachineTranslated'] = true
+                        node.textContent = response.data;
+                        node.classList.add("mojito-mt-ict-string-static");
+                        node.addEventListener("click", (e) => addMTCSS(e, node))
```

Comment:
i'm not sure to understand the event listeners below. Why do we need them?

Reply:
I had an issue where if you clicked an MT'd text unit than the CSS MT label was removed from the content, I suspect it's to do with React re-rendering the page which updates the content and removes the CSS appended 'MT' label